bobtimus/src/bin/bobtimus.rs

```diff
@@ -42,23 +41,19 @@ async fn main() -> Result<()> {
             };
             let bobtimus = Arc::new(Mutex::new(bobtimus));
 
-            match (tls_certificate, tls_private_key) {
-                (Some(certificate), Some(private_key)) => {
-                    warp::serve(http::routes(bobtimus, subscription))
-                        .tls()
-                        .cert_path(certificate)
-                        .key_path(private_key)
-                        .run(listen_address)
-                        .await;
-                }
-                (None, None) => {
-                    warp::serve(http::routes(bobtimus, subscription))
-                        .run(listen_address)
-                        .await;
-                }
-                _ => tracing::error!(
-                    "TLS certificate and private key have to be specified together."
-                ),
+            if let Some(https) = https {
+                warp::serve(http::routes(bobtimus.clone(), subscription.clone()))
+                    .tls()
+                    .cert_path(https.tls_certificate)
+                    .key_path(https.tls_private_key)
+                    .run(https.listen_https)
+                    .await;
+            }
+
+            if let Some(listen_http) = http {
+                warp::serve(http::routes(bobtimus, subscription))
+                    .run(listen_http)
+                    .await;
```

Comment:
Did you try if this now listens on both? 
I think `warp::serve` is blocking and once `if let Some(https) == Some` the part below will never be executed.

Reply:
I adapted the `bobtimus` code - did not bother with `fake_bobtimus` and added a `bail` for `https` instead.